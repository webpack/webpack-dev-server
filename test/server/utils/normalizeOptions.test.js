'use strict';

const path = require('path');
const webpack = require('webpack');
const normalizeOptions = require('../../../lib/utils/normalizeOptions');

describe('normalizeOptions', () => {
  const cases = [
    {
      title: 'no options',
      multiCompiler: false,
      options: {},
      optionsResults: null,
    },
    {
      title: 'transportMode sockjs string',
      multiCompiler: false,
      options: {
        transportMode: 'sockjs',
      },
      optionsResults: null,
    },
    {
      title: 'transportMode ws string',
      multiCompiler: false,
      options: {
        transportMode: 'ws',
      },
      optionsResults: null,
    },
    {
      title: 'transportMode ws object',
      multiCompiler: false,
      options: {
        transportMode: {
          server: 'ws',
          client: 'ws',
        },
      },
      optionsResults: null,
    },
    {
      title: 'transportMode custom server path',
      multiCompiler: false,
      options: {
        transportMode: {
          server: '/path/to/custom/server/',
        },
      },
      optionsResults: null,
    },
    {
      title: 'transportMode custom server class',
      multiCompiler: false,
      options: {
        transportMode: {
          server: class CustomServerImplementation {},
        },
      },
      optionsResults: null,
    },
    {
      title: 'transportMode custom client path',
      multiCompiler: false,
      options: {
        transportMode: {
          client: '/path/to/custom/client/',
        },
      },
      optionsResults: null,
    },
    {
      title: 'client host and port',
      multiCompiler: false,
      options: {
        client: {
          host: 'my.host',
          port: 9000,
        },
      },
      optionsResults: null,
    },
    {
      title: 'client path',
      multiCompiler: false,
      options: {
        client: {
          path: '/custom/path/',
        },
      },
      optionsResults: null,
    },
    {
      title: 'client path without leading/ending slashes',
      multiCompiler: false,
      options: {
        client: {
          path: 'custom/path',
        },
      },
      optionsResults: null,
    },
    {
      title: 'liveReload is true',
      multiCompiler: false,
      options: {
        liveReload: true,
      },
      optionsResults: null,
    },
    {
      title: 'liveReload is false',
      multiCompiler: false,
      options: {
        liveReload: false,
      },
      optionsResults: null,
    },
    {
      title: 'hot is true',
      multiCompiler: false,
      options: {
        hot: true,
      },
      optionsResults: null,
    },
    {
      title: 'hot is false',
      multiCompiler: false,
      options: {
        hot: false,
      },
      optionsResults: null,
    },
    {
      title: 'hot is only',
      multiCompiler: false,
      options: {
        hot: 'only',
      },
      optionsResults: null,
    },
    {
      title: 'dev is set',
      multiCompiler: false,
      options: {
        dev: {
          serverSideRender: true,
        },
      },
      optionsResults: null,
    },
    {
      title: 'static is true',
      multiCompiler: false,
      options: {
        static: true,
      },
      optionsResults: null,
    },
    {
      title: 'static is false',
      multiCompiler: false,
      options: {
        static: false,
      },
      optionsResults: null,
    },
    {
      title: 'static is string',
      multiCompiler: false,
      options: {
        static: '/static/path',
      },
      optionsResults: null,
    },
    {
      title: 'static is an array of strings',
      multiCompiler: false,
      options: {
        static: ['/static/path1', '/static/path2'],
      },
      optionsResults: null,
    },
    {
      title: 'static is an array of static objects',
      multiCompiler: false,
      options: {
        static: [
          {
            directory: '/static/path1',
          },
          {
            publicPath: '/static/public/path',
          },
        ],
      },
      optionsResults: null,
    },
    {
      title: 'static is an array of strings and static objects',
      multiCompiler: false,
      options: {
        static: [
          '/static/path1',
          {
            publicPath: '/static/public/path/',
          },
        ],
      },
      optionsResults: null,
    },
    {
      title: 'static is an object',
      multiCompiler: false,
      options: {
        static: {
          directory: '/static/path',
        },
      },
      optionsResults: null,
    },
    {
      title: 'static directory is an absolute url and throws error',
      multiCompiler: false,
      options: {
        static: {
          directory: 'http://localhost:8080',
        },
      },
      optionsResults: null,
      throws: 'Using a URL as static.directory is not supported',
    },
    {
      title: 'static publicPath is a string',
      multiCompiler: false,
      options: {
        static: {
          publicPath: '/static/public/path/',
        },
      },
      optionsResults: null,
    },
    {
      title: 'static publicPath is an array',
      multiCompiler: false,
      options: {
        static: {
          publicPath: ['/static/public/path1/', '/static/public/path2/'],
        },
      },
      optionsResults: null,
    },
    {
      title: 'static watch is false',
      multiCompiler: false,
      options: {
        static: {
          watch: false,
        },
      },
      optionsResults: null,
    },
    {
      title: 'static watch is true',
      multiCompiler: false,
      options: {
        static: {
          watch: true,
        },
      },
      optionsResults: null,
    },
    {
      title: 'static watch is an object',
      multiCompiler: false,
      options: {
        static: {
          watch: {
            poll: 500,
          },
        },
      },
      optionsResults: null,
    },
    {
      title: 'static serveIndex is false',
      multiCompiler: false,
      options: {
        static: {
          serveIndex: false,
        },
      },
      optionsResults: null,
    },
    {
      title: 'static serveIndex is true',
      multiCompiler: false,
      options: {
        static: {
          serveIndex: true,
        },
      },
      optionsResults: null,
    },
    {
      title: 'static serveIndex is an object',
      multiCompiler: false,
      options: {
        static: {
          serveIndex: {
            icons: false,
          },
        },
      },
      optionsResults: null,
    },

    {
      title: 'single compiler watchOptions is object',
      multiCompiler: false,
      options: {},
      optionsResults: null,
      webpackConfig: {
        watch: true,
        watchOptions: {
          aggregateTimeout: 300,
        },
      },
    },
    {
      title: 'single compiler watchOptions is object with watch false',
      multiCompiler: false,
      options: {},
      optionsResults: null,
      webpackConfig: {
        watch: false,
        watchOptions: {
          aggregateTimeout: 300,
        },
      },
    },
    {
      title: 'single compiler watchOptions is object with static watch true',
      multiCompiler: false,
      options: {
        static: {
          watch: true,
        },
      },
      optionsResults: null,
      webpackConfig: {
        watch: true,
        watchOptions: {
          aggregateTimeout: 300,
        },
      },
    },
    {
      title:
        'single compiler watchOptions is object with static watch overriding it',
      multiCompiler: false,
      options: {
        static: {
          watch: {
            poll: 500,
          },
        },
      },
      optionsResults: null,
      webpackConfig: {
        watch: true,
        watchOptions: {
          aggregateTimeout: 300,
        },
      },
    },
    {
      title: 'multi compiler watchOptions is set',
      multiCompiler: true,
      options: {},
      optionsResults: null,
      webpackConfig: [
        {},
        // watchOptions are set on the second compiler
        {
          watch: true,
          watchOptions: {
            aggregateTimeout: 300,
          },
        },
      ],
    },
    {
      title: 'firewall is set',
      multiCompiler: false,
      options: {
        firewall: false,
      },
      optionsResults: null,
    },
  ];

  cases.forEach((data) => {
    describe(data.title, () => {
      let compiler;
      beforeAll(() => {
        let webpackConfig;
        if (data.multiCompiler) {
          webpackConfig = require('../../fixtures/multi-compiler-config/webpack.config');
          if (Array.isArray(data.webpackConfig)) {
            webpackConfig = data.webpackConfig.map((config, index) =>
              Object.assign({}, webpackConfig[index], config)
            );
          }
        } else {
          webpackConfig = require('../../fixtures/simple-config/webpack.config');
          if (data.webpackConfig) {
            webpackConfig = Object.assign(
              {},
              webpackConfig,
              data.webpackConfig
            );
          }
        }

        compiler = webpack(webpackConfig);
      });

      it('should set correct options', () => {
        const originalContentBase = data.options.contentBase;
        if (data.throws) {
          expect(() => {
            normalizeOptions(compiler, data.options);
          }).toThrow();
          return;
        }

        normalizeOptions(compiler, data.options);
        if (data.optionsResults) {
          expect(data.options).toEqual(data.optionsResults);
        } else {
          if (data.options.contentBase !== originalContentBase) {
            // we don't want this in the snapshot, because it is
            // the current working directory
            expect(data.options.contentBase).toEqual(process.cwd());
            delete data.options.contentBase;
          }

          if (data.options.static) {
            data.options.static.forEach((staticOpts) => {
              if (staticOpts.directory === path.join(process.cwd(), 'public')) {
                // give an indication in the snapshot that this is the
                // current working directory
                staticOpts.directory = 'CWD';
              }
            });
          }

          expect(data.options).toMatchSnapshot();
        }
      });
    });
  });
});
