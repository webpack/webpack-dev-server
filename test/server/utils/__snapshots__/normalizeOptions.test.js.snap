// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeOptions client host and port should set correct options 1`] = `
Object {
  "client": Object {
    "host": "my.host",
    "path": "/ws",
    "port": 9000,
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions client path should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/custom/path",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions client path without leading/ending slashes should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/custom/path",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions dev is set should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {
    "serverSideRender": true,
  },
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions firewall is set should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": false,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions hot is false should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": false,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions hot is only should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": "only",
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions hot is true should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions liveReload is false should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": false,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions liveReload is true should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions multi compiler watchOptions is set should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {
        "aggregateTimeout": 300,
      },
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions no options should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions single compiler watchOptions is object should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {
        "aggregateTimeout": 300,
      },
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions single compiler watchOptions is object with static watch overriding it should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {
        "poll": 500,
      },
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions single compiler watchOptions is object with static watch true should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {
        "aggregateTimeout": 300,
      },
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions single compiler watchOptions is object with watch false should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is an array of static objects should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "/static/path1",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/static/public/path",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is an array of strings and static objects should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "/static/path1",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/static/public/path/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is an array of strings should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "/static/path1",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
    Object {
      "directory": "/static/path2",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is an object should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "/static/path",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is false should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": false,
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is string should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "/static/path",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static is true should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static publicPath is a string should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/static/public/path/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static publicPath is an array should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/static/public/path1/",
        "/static/public/path2/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static serveIndex is an object should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": false,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static serveIndex is false should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": false,
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static serveIndex is true should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static watch is an object should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {
        "poll": 500,
      },
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static watch is false should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": false,
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions static watch is true should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions transportMode custom client path should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "/path/to/custom/client/",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions transportMode custom server class should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": [Function],
  },
}
`;

exports[`normalizeOptions transportMode custom server path should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "/path/to/custom/server/",
  },
}
`;

exports[`normalizeOptions transportMode sockjs string should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "sockjs",
    "server": "sockjs",
  },
}
`;

exports[`normalizeOptions transportMode ws object should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;

exports[`normalizeOptions transportMode ws string should set correct options 1`] = `
Object {
  "client": Object {
    "path": "/ws",
  },
  "dev": Object {},
  "firewall": true,
  "hot": true,
  "liveReload": true,
  "static": Array [
    Object {
      "directory": "CWD",
      "publicPath": Array [
        "/",
      ],
      "serveIndex": Object {
        "icons": true,
      },
      "staticOptions": Object {},
      "watch": Object {},
    },
  ],
  "transportMode": Object {
    "client": "ws",
    "server": "ws",
  },
}
`;
